share/src/main/java/org/apache/kafka/server/share/persister/DefaultStatePersister.java

```diff
@@ -211,7 +228,8 @@ private ReadShareGroupStateResult readResponsesToResult(
                         int partition = partitionFuture.getKey();
                         CompletableFuture<ReadShareGroupStateResponse> future = partitionFuture.getValue();
                         try {
-                            ReadShareGroupStateResponse partitionResponse = future.get();
+                            // already completed because of allOf call in the caller
+                            ReadShareGroupStateResponse partitionResponse = future.get(0, TimeUnit.MILLISECONDS);
```

Comment:
Why not to combine future and call thenApply? Isn't thenApply will be triggered when all futures are completed?

Reply:
@apoorvmittal10 
Yes, this is what is happening the call to this util method happens like this:
`
CompletableFuture<Void> combinedFuture = CompletableFuture.allOf(...);
return combinedFuture.thenApply(v -> readResponsesToResult(futureMap));
`

Code ref: https://github.com/apache/kafka/blob/trunk/share/src/main/java/org/apache/kafka/server/share/persister/DefaultStatePersister.java#L195

Reply:
> allOff returns a future which will be completed when all underlying futures complete

Make sense then will it be better to just use `join()`? You won't need checked exceptions then rather unwrapping the CompletionException. As anyways the futures are completed. I leave it on you to decide.

Reply:
@apoorvmittal10 
`join` will not serve the purpose as it is a blocking call. The way we have implemented, there is no blocking at all until the final get call is made by the caller of the `DefaultStatePersister`. Join would block the response to caller.